Sources/DVR/SessionDataTask.swift

```diff
@@ -111,12 +119,17 @@ final class SessionDataTask: URLSessionDataTask {
                 }
             }
 
+            if filteredRequest == nil {
+                persistInteraction = false
+            }
+            
             if persistInteraction {
-                this.interaction = Interaction(request: filteredRequest, response: filteredResponse, responseData: filteredData)
-                this.session.finishTask(this, interaction: this.interaction!, playback: false)
+                this.interaction = Interaction(request: filteredRequest!, response: filteredResponse, responseData: filteredData)
+                if let interaction = this.interaction {
+                    this.session.finishTask(this, interaction: interaction, playback: false)
+                }
             } else {
-                this.interaction = Interaction(request: filteredRequest, response: response, responseData: data)
-                this.session.finishTask(this, interaction: this.interaction!, playback: true)
+                this.session.finishTask(this, responseData: filteredData, playback: true)
```

Comment:
If you follow the above suggestion and let `finishTask` take a nil interaction, I would recommend:

```swift
if persistInteraction {
  let interaction = Interaction(...)
  this.session.finishTask(this, interaction: interaction, playback: false)
} else {
  this.session.finishTask(this, interaction: nil, playback: true)
}
```

Reply:
It looks like you are either passing in an interaction (which will be persisted) or just data, which will be passed through. In that case, it might make sense to maintain two signatures, but you will want to (a) ensure they share as much common implementation as possible and (b) rename them so they are self documenting. 

Ex:
```
finishTaskAndPersist(this, interaction: interaction, ...)
finishTaskAndDoNotPersist(this, usingData: data, ...)
```

To me, the above names are clunky, but they are more self-documenting.